pkg/networkservice/connectioncontextkernel/ethernetcontext/server.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ethernetcontext provides networkservice chain elements for applying ethernet context
+package ethernetcontext
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connectioncontext"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservice/vppagent"
+)
+
+// NewServer creates a NetworkServiceServer chain element to set the EthernetContext for Kernel connection request
+func NewServer(сс *grpc.ClientConn) networkservice.NetworkServiceServer {
+	return &setEthernetContextServer{
+		client: configurator.NewConfiguratorClient(сс),
+	}
+}
+
+type setEthernetContextServer struct {
```

Comment:
setEthernetContextServer actual is a goal of the task. This server gets from vpp-agent mac of DST interface and puts it into ethernet context.

Reply:
Please *don't* combine getmac into a top level ethernetcontext/server.go ... we have no way of knowing a priori whether this is correct behavior or not in all cases that this is appropriate behavior.   Instead simply have a ethernetcontext/getmac/server.go that gets the mac address, and sets it to the dst mac address.